### PR TITLE
Implement `flaggedWalletModal`.

### PR DESCRIPTION
**Pull request by szilardszaloki**
Resolves https://github.com/brave/brave-browser/issues/21226.

flaggedWalletModal

`Learn more about Brave's process for flagging Rewards profiles.` links to https://support.brave.com/hc/en-us/articles/4494596374925 (not published yet).

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by emerick** [comment]
This modal seems kind of verbose. I totally get why, but I think it will have issues on mobile at least.

File: components/brave_rewards/resources/ui/components/modalRedirect/index.tsx [hunk 1]
```diff
@@ -18,8 +18,8 @@ import { getLocale, splitStringForTag } from '../../../../../common/locale'
 
 export interface Props {
   id?: string
-  errorText?: string
-  errorTextLink?: string
+  errorText?: string[]
+  errorTextLink?: string[]
```
Comment (emerick) [inline components/brave_rewards/resources/ui/components/modalRedirect/index.tsx:22]: I feel like there should only be one `errorTextLink`. Making it an array seems like overkill, unless there's a specific use case you have in mind.

**Comment by szilardszaloki** [comment]
> This modal seems kind of verbose. I totally get why, but I think it will have issues on mobile at least.

Yeah, probably. Maybe we need a somewhat more terse version for Android? cc: @Miyayes

**Comment by emerick** [comment]
@szilardszaloki @Miyayes Maybe the message can be crafted in such a way that the first two strings can be standalone on mobile and the next two strings add some extra context for desktop users? Just a thought!

File: components/brave_rewards/resources/ui/components/modalRedirect/index.tsx (hunk 1 above)
Comment (szilardszaloki) [reply components/brave_rewards/resources/ui/components/modalRedirect/index.tsx:22]: Usually we have one-paragraph texts, so as far as the usage, the extra brackets are not inconvenient, IMO.
E.g.:
```
errorText={[getLocale('redirectModalWalletOwnershipVerificationFailureText').replace('$1', getWalletProviderName(externalWallet))]}
errorTextLink={['https://community.brave.com']}
```
Now that we introduce multi-paragraph texts, I'm pretty sure we'll need a means of setting a separate link for each of them sooner or later. 🙂 That said, I can get rid of it if you find it unnecessary.

Comment (emerick) [reply components/brave_rewards/resources/ui/components/modalRedirect/index.tsx:22]: OK, gotcha. It's not a deal-breaker for me by any stretch. The only reason I brought it up is that this construction seems less than ideal (and would probably be the norm for the majority of these cases, I think):

https://github.com/brave/brave-core/pull/12456/files#diff-0c26c120bbe64c0639795a351794bad4c74ef608299086d608006fdfc7d320e6R239-R243

**Review by emerick — APPROVED** [review]
LGTM, listed one small nit/concern but it's not a deal-breaker for me.

File: components/brave_rewards/resources/ui/components/modalRedirect/index.tsx (hunk 1 above)
Comment (szilardszaloki) [reply components/brave_rewards/resources/ui/components/modalRedirect/index.tsx:22]: Oh, I see. I think we can clean up the syntax there a bit:
```
export interface Props {
  ...
  errorTextLink?: (string | undefined)[]
  ...
}

...

errorTextLink={[,,,'https://support.brave.com/hc/en-us/articles/4494596374925']}
```
WDYT?

Comment (emerick) [reply components/brave_rewards/resources/ui/components/modalRedirect/index.tsx:22]: Could it be `errorTextLink?: string | string[]`? I feel like that may be the simplest approach, assuming it's possible and doesn't over-complicate the code. @zenparsing might have some ideas here.

**Comment by Miyayes** [comment]
> @szilardszaloki @Miyayes Maybe the message can be crafted in such a way that the first two strings can be standalone on mobile and the next two strings add some extra context for desktop users? Just a thought!

If the user can scroll the modal on Android, I think it's OK to keep it this way (unless it gets cropped off or something and the user can't see longer texts). I think having all the information there slightly outweighs UI in this specific error case.

**Comment by emerick** [comment]
Screenshot_20220302-153621

@szilardszaloki @Miyayes Here's the screenshot of this modal in action on Android.

**Comment by szilardszaloki** [comment]
> @szilardszaloki @Miyayes Here's the screenshot of this modal in action on Android.

That's actually not that bad, IMO.

File: components/brave_rewards/resources/ui/components/modalRedirect/index.tsx (hunk 1 above)
Comment (zenparsing) [reply components/brave_rewards/resources/ui/components/modalRedirect/index.tsx:22]: So the current (undocumented) protocol here is that:

- `$1` is the provider name
- `$2` is the start of the learn more link
- `$3` is the end of the learn more link

My guess is that the fact that we have more than one line now doesn't really change this protocol, and we should probably just support the a single learn more link (which can be in any of the lines).

So my vote would be to leave it `errorTextLink?: string`.

Comment (szilardszaloki) [reply components/brave_rewards/resources/ui/components/modalRedirect/index.tsx]: Amended.

**Review by zenparsing — APPROVED** [review]

**Comment by szilardszaloki** [comment]
`SonarCloud` is reporting duplicated lines due to how we currently structure code in `rewards_reducer.ts`, `settingsPage.tsx` and `wallet_reducer.ts` (both on desktop and Android).

**Comment by LaurenWags** [comment]
Verification in progress with

```
Brave | 1.38.36 Chromium: 99.0.4844.51 (Official Build) nightly (x86_64)
-- | --
Revision | d537ec02474b5afe23684e7963d538896c63ac77-refs/branch-heads/4844@{#875}
OS | macOS Version 12.2.1 (Build 21D62)
```

Verification notes can be found under https://github.com/brave/internal/issues/855#issuecomment-1065157112

**Comment by srirambv** [comment]
Verification passed on Oppo Reno 5 with Android 12 running 1.38.37 x64 nightly build
- Verification notes can be found under https://github.com/brave/internal/issues/855#issuecomment-1065261148

**Pull request merged**